cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -97,6 +100,130 @@ func recoverySystemEssentialSnaps(seedDir, recoverySystem string, essentialTypes
 // generateMountsMode* is called multiple times from initramfs until it
 // no longer generates more mount points and just returns an empty output.
 func generateMountsModeInstall(recoverySystem string) error {
+	var buf bytes.Buffer
```

Comment:
I understand that this makes for a smaller change, but is a weird API for this because is very unstructured.

I think the helper should return something like   (mountReqs []string, err)  instead

Reply:
or return a flag whether it mounted all it's responsible for

Reply:
Thanks, I changed this so that there is flag returned now.

Reply:
Changed this in #8010 fwiw, will merge here once we are happy with 8010.